frontend/packages/console-app/src/actions/creators/common-factory.ts

```diff
@@ -0,0 +1,116 @@
+import i18next from 'i18next';
+import { Action } from '@console/dynamic-plugin-sdk';
+import {
+  annotationsModal,
+  deleteModal,
+  labelsModal,
+  configureReplicaCountModal,
+  podSelectorModal,
+  tolerationsModal,
+} from '@console/internal/components/modals';
+import { resourceObjPath, asAccessReview } from '@console/internal/components/utils';
+import { referenceForModel, K8sKind, K8sResourceKind } from '@console/internal/module/k8s';
+
+export type ResourceActionCreator = (
+  kind: K8sKind,
+  obj: K8sResourceKind,
+  relatedResource?: K8sResourceKind,
+) => Action;
+
+export type ResourceActionFactory = { [name: string]: ResourceActionCreator };
+
+export const CommonActionFactory: ResourceActionFactory = {
+  Delete: (kind: K8sKind, obj: K8sResourceKind): Action => ({
+    id: `delete-${kind.kind}-action`,
```

Comment:
I'd expect a consistent ID.
Doesn't add much value to add the `-action` suffix to every single action.
```suggestion
    id: `delete-resource`,
```

Reply:
Updated.